index.js

```diff
@@ -1,36 +1,93 @@
+// this entire module is depressing.
+
 const os = require('os');
 const fs = require('fs');
 const path = require('path');
 const execFile = require('child_process').execFile;
-
-const tmpfile = path.join(os.tmpdir(), process.versions.v8+'.flags.json');
+const env = process.env;
+const user = env.LOGNAME || env.USER || env.LNAME || env.USERNAME;
+const configfile = '.v8flags.'+process.versions.v8+'.'+user+'.json';
 const exclusions = ['--help'];
 
-module.exports = function (cb) {
-  try {
-    var flags = require(tmpfile);
-    process.nextTick(function(){
-      cb(null, flags);
-    });
-  } catch (e) {
-    execFile(process.execPath, ['--v8-options'], function (execErr, result) {
-      var flags;
-      if (execErr) {
-        return cb(execErr);
-      }
-      flags = result.match(/\s\s--(\w+)/gm).map(function (match) {
-        return match.substring(2);
-      }).filter(function (name) {
-        return exclusions.indexOf(name) === -1;
+// in some cases, a user's home directory can't be found, nor can a temporary
+// directory. (>_<) in cases where everything fails, fall back to using the
+// same directory as this file. hopefully the process running this has access
+// to write to it.
+function openConfig (cb) {
+  var userHome = require('user-home');
+  var dir =  path.join(userHome || os.tmpdir(), configfile);
+  fs.open(dir, 'a+', function (firstOpenErr, firstFd) {
```

Comment:
Okay yeah, this is fine.  If the file has a size larger than 0 it will never be written to.